src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -334,15 +334,64 @@ else if (!period.isEmpty()) {
      *         a summary tag.
      */
     private static Optional<DetailNode> getInlineSummaryTag(DetailNode javadoc) {
-        Optional<DetailNode> node = Arrays.stream(javadoc.getChildren())
+        return Arrays.stream(javadoc.getChildren())
                 .filter(SummaryJavadocCheck::isInlineTagPresent)
                 .findFirst()
-                .map(SummaryJavadocCheck::getInlineTagNodeWithinHtmlElement);
+                .map(SummaryJavadocCheck::getInlineTagNodeWithinHtmlElement)
+                .filter(node -> isSummaryTag(node) && isDefinedFirst(node));
+    }
 
-        if (node.isPresent() && !isSummaryTag(node.get())) {
-            node = Optional.empty();
+    /**
+     * Whether the {@code {@summary}} tag is defined first in the javadoc.
+     *
+     * @param inlineSummaryTag node of type {@link JavadocTokenTypes#JAVADOC_INLINE_TAG}
+     * @return {@code true} if the {@code {@summary}} tag is defined first in the javadoc
+     */
+    private static boolean isDefinedFirst(DetailNode inlineSummaryTag) {
+        boolean isDefinedFirst = true;
+        DetailNode previousSibling = JavadocUtil.getPreviousSibling(inlineSummaryTag);
+        while (previousSibling != null && isDefinedFirst) {
+            final int siblingType = previousSibling.getType();
+            if (siblingType == JavadocTokenTypes.TEXT) {
+                if (!previousSibling.getText().isBlank()) {
+                    isDefinedFirst = false;
+                }
+            }
+            else if (siblingType == JavadocTokenTypes.HTML_ELEMENT
+                && isTextPresentInsideHtmlTag(previousSibling)) {
+                isDefinedFirst = false;
+            }
+            previousSibling = JavadocUtil.getPreviousSibling(previousSibling);
+        }
+        return isDefinedFirst;
+    }
+
+    /**
+     * Whether some text is present inside the HTML element or tag.
+     *
+     * @param node DetailNode of type {@link JavadocTokenTypes#HTML_TAG}
+     *             or {@link JavadocTokenTypes#HTML_ELEMENT}
+     * @return {@code true} if some text is present inside the HTML element or tag
+     */
+    public static boolean isTextPresentInsideHtmlTag(DetailNode node) {
+        DetailNode nestedChild = JavadocUtil.getFirstChild(node);
+        if (node.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+            nestedChild = JavadocUtil.getFirstChild(nestedChild);
+        }
+        boolean isTextPresentInsideHtmlTag = false;
+        while (nestedChild != null) {
+            final int nestedChildType = nestedChild.getType();
+            if (nestedChildType == JavadocTokenTypes.TEXT
+                && !nestedChild.getText().isBlank()
+                || (nestedChildType == JavadocTokenTypes.HTML_TAG
+                    || nestedChildType == JavadocTokenTypes.HTML_ELEMENT)
+                && isTextPresentInsideHtmlTag(nestedChild)) {
+                isTextPresentInsideHtmlTag = true;
+                break;
```

Comment:
to make it consistent with previous method's style (no break), it can be
```
boolean isTextPresent = false;
while (nestedChild != null && !isTextPresent) {
   final int nestedChildType = nestedChild.getType();
   isTextPresent = nestedChildType == JavadocTokenTypes.TEXT
                && !nestedChild.getText().isBlank()
                || (nestedChildType == JavadocTokenTypes.HTML_TAG
                    || nestedChildType == JavadocTokenTypes.HTML_ELEMENT
}
```

what do you think?

Reply:
Good suggestion, done.